### ISSUE TITLE
Create `FriendlyCaptcha` 

### ISSUE THREAD

**Issue by RyanPaulGannon** [activist-org/activist#369]
### Terms

- [X] I have searched [open and closed design issues](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Adesign)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

The issue is to create the friendly captcha, see this [link](https://friendlycaptcha.com) for info. 

An example of this component can be seen below:

![](https://user-images.githubusercontent.com/24387426/260310858-0241d107-2977-45b6-92c1-1ee0bea423ac.png)

### Contribution

I'll look in to this, if anyone would like to help then hit me up :)

**Comment by andrewtavis** [comment]
Thanks for the issue, @RyanPaulGannon! Will assign you, but we can also be open to others wanting to contribute :)

**Comment by andrewtavis** [comment]
Btw, let me just say that I really appreciate the quality of this issue here! 🥳 A link, screenshot and welcoming attitude to people joining in is so great to see. I don’t be surprised if I ask you to make some more 😅😊😊

**Comment by RyanPaulGannon** [comment]
Partly pinched most of yours from the PR I'm on with 😅, and also [these guys'](https://github.com/RIP-Comm/clementine) formatting, I thought is really quite tidy too!

**Comment by llambrecht** [comment]
Hey ! I would love to contribute to this project. Let me know if I can help on this issue 😊

**Comment by andrewtavis** [comment]
Hey @llambrecht! Thanks for jumping into so many issues right away! Really great 😊 I’ll assign you as well and maybe you can start with some research, then you and @RyanPaulGannon can discuss who makes the change :)

**Comment by llambrecht** [comment]
Hey,
Just letting you know that I haven't had the time to look closely at this yet but I will get to it soon !

**Comment by andrewtavis** [comment]
100% ok, @llambrecht :) Could also be something that you work on this weekend for Code Night 🌙  if you were thinking of joining us on Saturday (dev team is getting together every two months for a day of work) 😊 Would be great to have you!

**Comment by andrewtavis** [comment]
Appreciate the communication on this as well! :)

**Comment by llambrecht** [comment]
> 100% ok, @llambrecht :) Could also be something that you work on this weekend for Code Night 🌙 if you were thinking of joining us on Saturday (dev team is getting together every two months for a day of work) 😊 Would be great to have you!

Not available on saturday but i'll try to be for the next one ! :)

**Comment by llambrecht** [comment]
Hey @andrewtavis @RyanPaulGannon ,

I started looking into this.
Friendly captcha has a free version for non-commercial project that allows 1000 request per month. Is that enough ?

In which form(s) do you want to include the captcha in ? The contact form (/help/contact) ? Others ?

**Comment by andrewtavis** [comment]
Hey @llambrecht! Thanks for the research here :) 1000 per month will certainly be enough for the start, and then we can start paying for it later. 

Aside from the contact page, the other one for now would be [activist.org/auth/sign-in/](https://activist.org/auth/sign-in/), which actually at this point has a pretty solid component made for this, although we can of course switch it as needed. We'll also have sign up soon as well, so those three pages 😊

**Comment by andrewtavis** [comment]
Hey there @llambrecht 👋 Doing a quick checkin on issues and wanted to see how you're doing here :) Let me know if there's anything I can do to help!

**Comment by llambrecht** [comment]
Hey,
Sorry for the lack of updates. I've been very busy lately. I'll try to make a PR next week. However, if I'm unable to do so, i'll let someone else work on this to avoid blocking.
I'll contact you next week if I need help if thats ok for you !

**Comment by andrewtavis** [comment]
Yes by all means, @llambrecht! Take your time, and thanks for the work you've put in already!

**Comment by llambrecht** [comment]
Apologies for not being able to address this sooner. I'll unassign myself for now, in case someone else is available and would like to take it over.

**Comment by andrewtavis** [comment]
No stress, @llambrecht! Hope you're well! Please let us know if you want to pick up another issue in the future!

**Comment by AkshithaReddy1899** [comment]
Hi @andrewtavis I am willing to take this issue, can you please assign to me

**Comment by andrewtavis** [comment]
Very happy to, @AkshithaReddy1899! Please let me know if there's anything we can do to help 😊

**Comment by AkshithaReddy1899** [comment]
Thank you so much @andrewtavis.

To start with I want to make sure that I am on the right path, so please correct me if I got anything wrong

1. You want a captcha check on three pages signup, sign-in, contact
2. I have to use [this](https://friendlycaptcha.com/) to generate a captcha, in that case, I want someone to create an account and give me the API key (secret) and sitekey as mentioned [here](https://docs.friendlycaptcha.com/#/installation)

and also let me know if there is anything else to add 

Thank you.

**Comment by andrewtavis** [comment]
Thank you so much, @AkshithaReddy1899! Really appreciate the detail here as it makes it a lot easier to jump back into this with you. I'll sign up for friendly captcha later today and send along the needed info 😊 We'd need to store the secret/sitekey in a way that it's not visible in the code, correct? Maybe this is our first use of GitHub secrets?! :)

**Comment by AkshithaReddy1899** [comment]
Thank you @andrewtavis. let me know once the sign-up is done, so that I can get started

**Comment by andrewtavis** [comment]
Hey @AkshithaReddy1899 👋 Sorry for the wait on this! Has been a busy start to the year for me. I'm all signed up to Friendly Captcha now. Sending you an email with the API key and Sitekey for testing purposes. We can check them out ourselves with the ones I've made, and then I'll remake them and switch it into the .env file once we're in production 😊

Let me know if you need anything else here!

**Comment by andrewtavis** [comment]
#681 closes this up 🚀 Thanks so much for the work here, @AkshithaReddy1899! We'll have a bit more work to do once we deploy to get the challenge working as @AkshithaReddy1899 did the research that we'll obviously need to not only server side so that people can't disable it.

Thanks all so much for the conversation and work here! 🙏
